### PR TITLE
(FACT-969) Implement networking resolver for AIX

### PR DESCRIPTION
**Pull request by branan**
The AIX networking stack forked from BSD a very long time ago, and
interactions with it are kind of gross.

This is based on what little documentation IBM provides, old BSD
mailing list posts, the source code of 15-year old network daemons,
and a little bit of reverse engineering. It's as close to sane as
we're going to get working with these APIs.

The big grossness with this is that the APIs for the stack are
basically "stick a bunch of structs back to back with size
fields". This requires us to walk the buffer ourselves, parsing out
whatever we find based on the type/size fields.

### CODE REVIEW

**Comment by branan** [comment]
I want to do a pass on the solaris, posix, and BSD resolvers to see if there are any other helpers I could be using to process these structures in AIX, but the overall shape of the code is done here.

File: lib/src/facts/aix/networking_resolver.cc [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+            }
+        }
+
+        data.primary_interface = get_primary_interface();
+
+        return data;
+    }
+
+    networking_resolver::ndd_map networking_resolver::get_ndd_info() const {
+        int ksize = getkerninfo(KINFO_NDD, nullptr, nullptr, 0);
```
Comment (MikaelSmith) [inline lib/src/facts/aix/networking_resolver.cc]: This information could change between calls. Other places where we deal with this we've tended to loop several times in case the number of devices increased between calls. We should probably do that here.

File: lib/src/facts/aix/networking_resolver.cc [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        ndd_map result;
+        for (auto& device : ndd) {
+            result[device.ndd_name] = device;
+            result[device.ndd_alias] = device;
+        }
+        return result;
+    }
+
+    std::vector<networking_resolver::interface> networking_resolver::get_interfaces() const {
+        int ksize = getkerninfo(KINFO_RT_IFLIST, nullptr, nullptr, 0);
```
Comment (MikaelSmith) [inline lib/src/facts/aix/networking_resolver.cc]: Same comment here about looping to allow for system changes.

Comment (MikaelSmith) [inline lib/src/facts/aix/networking_resolver.cc]: You could probably use a templated helper to wrap getkerninfo with retries.

**Comment by MikaelSmith** [comment]
Everything else looks reasonable.

**Comment by MikaelSmith** [comment]
Can you retarget this at stable when you update it?

**Comment by branan** [comment]
closing this to retarget at stable
